go/interpreter/lfvm/interpreter_test.go

```diff
@@ -699,6 +628,152 @@ func BenchmarkSatisfiesStackRequirements(b *testing.B) {
 	}
 }
 
+////////////////////////////////////////////////////////////////////////////////
+// test utilities
+
+func Test_generateCodeForOps(t *testing.T) {
+	tests := map[OpCode]int{
+		PUSH1:                     1,
+		PUSH2:                     1,
+		PUSH3:                     2,
+		PUSH4:                     2,
+		PUSH5:                     3,
+		PUSH6:                     3,
+		PUSH31:                    16,
+		PUSH32:                    16,
+		PUSH1_PUSH1:               1,
+		PUSH1_PUSH4_DUP3:          3,
+		PUSH1_PUSH1_PUSH1_SHL_SUB: 2,
+	}
+	for op, test := range tests {
+		t.Run(op.String(), func(t *testing.T) {
+			code := generateCodeFor(op)
+			if want, got := test, len(code); want != got {
+				t.Errorf("expected %d instructions, got %d", want, got)
+			}
+		})
+	}
+}
+
+// generateCodeFor generates valid LFVM code for one instruction.
+// Appends necessary DATA instructions to the code to satisfy stack requirements.
+// Adds JUMPDEST instruction after JUMP instructions.
+func generateCodeFor(op OpCode) Code {
+
+	var code []Instruction
+
+	switch op {
+	case PUSH1_PUSH4_DUP3:
+		code = append(code, Instruction{op, 0}, Instruction{DATA, 0})
+	case PUSH1_PUSH1_PUSH1_SHL_SUB:
+		code = append(code, Instruction{op, 0}, Instruction{DATA, 0})
+	case PUSH2_JUMP:
+		code = append(code, Instruction{op, 1}) // hardcoded jump destination
+	case PUSH2_JUMPI:
+		code = append(code, Instruction{op, 1}) // hardcoded jump destination
+	default:
+		code = append(code, Instruction{op, 0})
+	}
+
+	for _, op := range append(op.decompose(), op) {
+		if PUSH3 <= op && op <= PUSH32 {
+			n := int(op) - int(PUSH3) + 3
+			numInstructions := n/2 + n%2
+			for i := 0; i < numInstructions-1; i++ {
+				code = append(code, Instruction{DATA, 0})
+			}
+		}
+	}
+
+	if isJump(op) {
+		code = append(code, Instruction{JUMPDEST, 0})
+	}
+
+	if op == JUMP_TO {
+		// prevent endless loop by having jump to itself
+		code[0].arg = uint16(len(code))
+		code = append(code, Instruction{JUMPDEST, 0})
+	}
+
+	return code
+}
+
+// fillStackFor fills the stack with the required number of elements for the given opcode.
+// For Jump instructions, it also encodes the PC for the the first jump destination found in code
+func fillStackFor(op OpCode, stack *stack, code Code) error {
+	limits := _precomputedStackLimits.get(op)
+	stack.stackPointer = limits.min
+
+	// jump instructions need a valid jump destination
+	if isJump(op) {
+		counter := slices.IndexFunc(code, func(v Instruction) bool {
+			return v.opcode == JUMPDEST
+		})
+		if counter == -1 {
+			return fmt.Errorf("missing JUMPDEST instruction")
+		}
+
+		for i := 0; i < limits.min; i++ {
+			stack.data[i] = *uint256.NewInt(uint64(counter))
+		}
+	}
+
+	return nil
+}
+
+var _isUndefinedOpCodeRegex = regexp.MustCompile(`^op\(0x[0-9A-Fa-f]+\)$`)
+
+func isExecutable(op OpCode) bool {
```

Comment:
would it be terrible if these 2 are inlined ?

Reply:
The reason why I did it this way is that I would not like to have a regex compiled on each loop iteration. Although it is test code. 
I refactored it, and now it is part of the function.

Reply:
I thought of not instantiating regex, but my thought as well was "this is not a performance critical function"